ethereum/contracts/zksync/Verifier.sol

```diff
@@ -1617,10 +1647,13 @@ contract Verifier is IVerifier {
                 pointMulAndAddIntoDest(PROOF_OPENING_PROOF_AT_Z_OMEGA_X_SLOT, u, PAIRING_PAIR_WITH_X_X_SLOT)
                 pointNegate(PAIRING_PAIR_WITH_X_X_SLOT)
 
-                // Add recursive proof part
-                let uu := mulmod(u, u, R_MOD)
-                pointMulAndAddIntoDest(PROOF_RECURSIVE_PART_P1_X_SLOT, uu, PAIRING_PAIR_WITH_GENERATOR_X_SLOT)
-                pointMulAndAddIntoDest(PROOF_RECURSIVE_PART_P2_X_SLOT, uu, PAIRING_PAIR_WITH_X_X_SLOT)
+                // Add recursive proof part if needed
+                let use_recursive := mload(VK_RECURSIVE_FLAG)
+                if not(iszero(use_recursive)) {
```

Comment:
`not` reverse all bits from 1 to 0 and 0 to 1. So: 
- if `use_recursive` is 0 then `not(iszero(0))` == `not(1)` == 0xfffffff...
- if `use_recursive` is 1 then `not(iszero(1))` == `not(0)` == 0xfffffff...

Reply:
Is it ok to write:
```    
switch use_recursive
case 0 {}
default {
    ...
}
```

Reply:
you can write 

```
if use_recursive {
...
}
```

`if` statement works on any non zero value